src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -559,6 +559,234 @@ await invokeTask.NoThrowAwaitable(); // may or may not throw due to cancellation
         await Assert.ThrowsAsync<RemoteMethodNotFoundException>(() => this.clientRpc.InvokeAsync<string>(nameof(Server.TestInvalidMethod)));
     }
 
+    [Fact]
+    public async Task InvokeAsync_InvokesMethodWithAttributeSet()
```

Comment:
I love the tests you're adding. I think we have a substantial set here now that it's probably worth moving them to their own file.